File: todo-back/src/main/java/ru/sergo/todoapp/todoback/dto/Response.java
package ru.sergo.todoapp.todoback.dto;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.experimental.UtilityClass;
import org.springframework.http.HttpStatus;

@Data
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class Response<T> {

    private Integer code;
    private T data;
    private String message;

    @UtilityClass
    public class ResponseBuilder {
        public <T> Response<T> ok(T data) {
            return new Response<>(HttpStatus.OK.value(), data, HttpStatus.OK.getReasonPhrase());
        }

        public <T> Response<T> fail(HttpStatus status, String message) {
            return new Response<>(status.value(), null, message);
        }
    }
}

File: expenses-calculation-service/ecs-back/src/main/kotlin/ru/sergo/ecs/ecsback/dto/Response.kt
package ru.sergo.ecs.ecsback.dto

import org.springframework.http.HttpStatus

data class Response<T> (val code: Int, val data: T?, val message: String) {

    companion object ResponseBuilder {
        fun <T> ok(data: T): Response<T> {
            return Response(HttpStatus.OK.value(), data, HttpStatus.OK.reasonPhrase)
        }

        fun <T> fail(status: HttpStatus , message: String): Response<T> {
            return Response(status.value(), null, message)
        }
    }
}